app/shared/types/pageSchema.ts

```diff
@@ -0,0 +1,74 @@
+import Ajv from 'ajv';
+
+import templatesModel from 'api/templates/templatesModel';
+
+import { objectIdSchema } from 'shared/types/commonSchemas';
+import { wrapValidator } from 'shared/tsUtils';
+import { PageType } from './pageType';
+
+export const emitSchemaTypes = true;
+
+const ajv = Ajv({ allErrors: true });
+
+ajv.addKeyword('validatePageIsNotEntityView', {
+  async: true,
+  errors: true,
+  type: 'object',
+  async validate(_fields: any, page: PageType) {
+    if (page.sharedId) {
+      const templates = await templatesModel.get({
+        entityViewPage: page.sharedId,
+      });
+
+      if (templates.length > 0 && !page.entityView) {
+        const templatesTitles = templates.map(template => template.name);
+        throw new Ajv.ValidationError([
+          {
+            keyword: 'validatePageIsNotEntityView',
+            schemaPath: '',
+            params: { keyword: 'validatePageIsEntityView', _fields },
+            message: `This page is in use by the following templates: ${templatesTitles.join(
+              ', '
+            )}`,
+            dataPath: '.pages',
+          },
+        ]);
+      }
+    }
+    return true;
+  },
+});
```

Comment:
i know there is already some schema validators that include async data consistency validations, but this one is new, can we leave it as only schema validator ? it makes sense to have this 2 concepts separated ? schema validator is agnostic and can be used in other contexts, like in the browser application.

Reply:
I am confused.  We have both separated: the file exports the agnostic schema validator, but it also exports the function to validate the page.  Is this not what you refer to?

Reply:
ok, understood this works for now, but the validator exported includes async validations, you are just exporting the schema as a separated export, to be used by the "route validator" right ?

Reply:
Yes, and client type definitions.